packages/artifact/src/internal/upload/upload-artifact.ts

```diff
@@ -28,11 +30,14 @@ export async function uploadArtifact(
     }
   }
 
-  // TODO - Implement upload functionality
+  const zipUploadStream = await createZipUploadStream(zipSpecification)
+
+  const sasURL = '' // TODO replace with actual URL
+  const uploadResult = await uploadZipToBlobStorage(sasURL, zipUploadStream)
```

Comment:
We should check whether the upload succeeded here, otherwise it might cause an error while trying to access `uploadResult.uploadSize` if an error occurs:
https://github.com/actions/toolkit/blob/1573d74a9f68f9b53cbac68264702ae9b9c9afb0/packages/artifact/src/internal/upload/blob-upload.ts#L51-L56